File: articles/institutional-change-and-behavioral-adaptation/cpp/institutional_change_score.cpp
#include <iostream>

struct InstitutionalChangeCase {
    double feedback_quality;
    double adaptive_capacity;
    double legitimacy;
    double incentive_alignment;
    double normative_support;
    double governance_capacity;
    double path_dependence;
    double behavioral_flexibility;
    double coordination_quality;
    double environmental_change;
    double distributional_attention;
    double transition_burden;
};

double institutional_change_score_raw(const InstitutionalChangeCase& x) {
    return
        0.13 * x.feedback_quality +
        0.14 * x.adaptive_capacity +
        0.10 * x.legitimacy +
        0.10 * x.incentive_alignment +
        0.09 * x.normative_support +
        0.12 * x.governance_capacity -
        0.12 * x.path_dependence +
        0.10 * x.behavioral_flexibility +
        0.08 * x.coordination_quality +
        0.06 * x.environmental_change +
        0.05 * x.distributional_attention -
        0.05 * x.transition_burden;
}

int main() {
    InstitutionalChangeCase demo {80, 78, 75, 70, 72, 82, 40, 76, 74, 65, 70, 35};
    std::cout << "Institutional change raw score: "
              << institutional_change_score_raw(demo) << std::endl;
    return 0;
}

File: articles/institutional-change-and-behavioral-adaptation/go/institutional_change_score.go
package main

import "fmt"

type InstitutionalChangeCase struct {
	FeedbackQuality        float64
	AdaptiveCapacity       float64
	Legitimacy             float64
	IncentiveAlignment     float64
	NormativeSupport       float64
	GovernanceCapacity     float64
	PathDependence         float64
	BehavioralFlexibility  float64
	CoordinationQuality    float64
	EnvironmentalChange    float64
	DistributionalAttention float64
	TransitionBurden       float64
}

func InstitutionalChangeScoreRaw(x InstitutionalChangeCase) float64 {
	return 0.13*x.FeedbackQuality +
		0.14*x.AdaptiveCapacity +
		0.10*x.Legitimacy +
		0.10*x.IncentiveAlignment +
		0.09*x.NormativeSupport +
		0.12*x.GovernanceCapacity -
		0.12*x.PathDependence +
		0.10*x.BehavioralFlexibility +
		0.08*x.CoordinationQuality +
		0.06*x.EnvironmentalChange +
		0.05*x.DistributionalAttention -
		0.05*x.TransitionBurden
}

func main() {
	demo := InstitutionalChangeCase{80, 78, 75, 70, 72, 82, 40, 76, 74, 65, 70, 35}
	fmt.Printf("Institutional change raw score: %.2f\n", InstitutionalChangeScoreRaw(demo))
}
